agent/util/subtree_unix.go

```diff
@@ -28,7 +31,11 @@ func TrackProcess(key string, pid int, logger grip.Journaler) {}
 
 // KillSpawnedProcs kills processes that descend from the agent and waits
 // for them to terminate.
-func KillSpawnedProcs(ctx context.Context, key, workingDir string, logger grip.Journaler) error {
+func KillSpawnedProcs(ctx context.Context, key, workingDir, execUser string, logger grip.Journaler) error {
+	if execUser != "" {
```

Comment:
Somewhere I think we should explain why we do this differently for certain cases, either a small comment here with some of the reasoning about macos, or [an ADR](https://docs.devprod.prod.corp.mongodb.com/evergreen/decisions) with some info from the comment (if the ADR, it could be fine to wait for the change to bake in a bit first)

Reply:
done and done